Datarace around LogSeriesLimiter and slowDownAfterRetryableError

### Describe the bug

https://pastila.nl/?00037689/45a51f1980f77091e40c279cd6b6e878#KDaEIV3eh2musH6bFBjA4A==

```
E           Exception: Sanitizer assert found for instance ==================
E           WARNING: ThreadSanitizer: data race (pid=727)
E             Read of size 8 at 0x721401987ad8 by thread T721:
E               #0 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::__get_long_size[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/string:1969:29 (clickhouse+0x130139d4) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #1 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::size[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/string:1284:26 (clickhouse+0x130139d4)
E               #2 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::empty[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/string:1324:12 (clickhouse+0x130139d4)
E               #3 LogSeriesLimiter::log(Poco::Message&&) ci/tmp/build/./src/Common/LoggingFormatStringHelpers.cpp:157:24 (clickhouse+0x130139d4)
E               #4 DB::S3::Client::slowDownAfterRetryableError() const ci/tmp/build/./src/IO/S3/Client.cpp:828:17 (clickhouse+0x19769e6a) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #5 std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const::'lambda'(DB::S3::UploadPartRequest const&)::operator()(DB::S3::UploadPartRequest const&) const ci/tmp/build/./src/IO/S3/Client.cpp:712:13 (clickhouse+0x19762df5) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #6 std::__1::invoke_result<DB::S3::UploadPartRequest, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequest<DB::S3::UploadPartRequest, std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const::'lambda'(DB::S3::UploadPartRequest const&)>(DB::S3::UploadPartRequest&, DB::S3::UploadPartRequest) const ci/tmp/build/./src/IO/S3/Client.cpp:639:23 (clickhouse+0x19762df5)
E               #7 std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const ci/tmp/build/./src/IO/S3/Client.cpp:771:12 (clickhouse+0x19762df5)
E               #8 DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const ci/tmp/build/./src/IO/S3/Client.cpp:548:12 (clickhouse+0x19762df5)
E               #9 DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0::operator()() const ci/tmp/build/./src/IO/WriteBufferFromS3.cpp:583:36 (clickhouse+0x198d6d86) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #10 decltype(std::declval<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>(DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x198d6d86)
E               #11 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>(DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x198d6d86)
E               #12 std::__1::__function::__default_alloc_func<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0, void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x198d6d86)
E               #13 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x198d6d86)
E               #14 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x19506ecf) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #15 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x19506ecf)
E               #16 DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0::operator()() ci/tmp/build/./src/Common/ThreadPoolTaskTracker.cpp:130:9 (clickhouse+0x19506ecf)
E               #17 decltype(std::declval<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>(DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x19506ecf)
E               #18 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>(DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x19506ecf)
E               #19 std::__1::__function::__default_alloc_func<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0, void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x19506ecf)
E               #20 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x19506ecf)
E               #21 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x155c1162) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #22 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x155c1162)
E               #23 std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()::operator()() ci/tmp/build/./src/Common/threadPoolCallbackRunner.h:45:20 (clickhouse+0x155c1162)
E               #24 decltype(std::declval<void>()()) std::__1::__invoke[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()&>(void&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x155c1162)
E               #25 std::__1::__packaged_task_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'(), std::__1::allocator<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()>, void ()>::operator()() ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1438:10 (clickhouse+0x155c1162)
E               #26 std::__1::__packaged_task_function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1589:10 (clickhouse+0x155c1494) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #27 std::__1::packaged_task<void ()>::operator()() ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1752:5 (clickhouse+0x155c1494)
E               #28 std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()::operator()() const ci/tmp/build/./src/Common/threadPoolCallbackRunner.h:52:70 (clickhouse+0x155c13ac) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #29 decltype(std::declval<void>()()) std::__1::__invoke[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&>(void&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x155c13ac)
E               #30 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&>(std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x155c13ac)
E               #31 std::__1::__function::__default_alloc_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'(), void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x155c13ac)
E               #32 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'(), void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x155c13ac)
E               #33 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x1313e25a) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #34 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x1313e25a)
E               #35 ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::worker() ci/tmp/build/./src/Common/ThreadPool.cpp:802:17 (clickhouse+0x1313e25a)
E               #36 decltype(*std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&>().*std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&, void>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x13146cc4) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #37 decltype(auto) std::__1::__apply_tuple_impl[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, 0ul>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<0ul>) ci/tmp/build/./contrib/llvm-project/libcxx/include/tuple:1354:5 (clickhouse+0x13146cc4)
E               #38 decltype(auto) std::__1::apply[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&) ci/tmp/build/./contrib/llvm-project/libcxx/include/tuple:1358:5 (clickhouse+0x13146cc4)
E               #39 ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()::operator()() ci/tmp/build/./src/Common/ThreadPool.h:312:13 (clickhouse+0x13146cc4)
E               #40 decltype(std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)()>()(std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>())) std::__1::__invoke[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x13146be1) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #41 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x13146be1)
E               #42 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x13146be1)
E               #43 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x13146be1)
E               #44 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x1313a722) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #45 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x1313a722)
E               #46 ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::worker() ci/tmp/build/./src/Common/ThreadPool.cpp:812:17 (clickhouse+0x1313a722)
E               #47 decltype(*std::declval<ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>().*std::declval<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, void>(void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x131437db) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #48 void std::__1::__thread_execute[abi:ne190107]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<2ul>) ci/tmp/build/./contrib/llvm-project/libcxx/include/__thread/thread.h:192:3 (clickhouse+0x131437db)
E               #49 void* std::__1::__thread_proxy[abi:ne190107]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>>(void*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__thread/thread.h:201:3 (clickhouse+0x131437db)
E           
E             Previous write of size 8 at 0x721401987ad8 by thread T733:
E               #0 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::__set_long_size[abi:ne190107](unsigned long) ci/tmp/build/./contrib/llvm-project/libcxx/include/string:1966:30 (clickhouse+0x13013a36) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #1 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::clear[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/string:3290:5 (clickhouse+0x13013a36)
E               #2 LogSeriesLimiter::log(Poco::Message&&) ci/tmp/build/./src/Common/LoggingFormatStringHelpers.cpp:160:23 (clickhouse+0x13013a36)
E               #3 DB::S3::Client::slowDownAfterRetryableError() const ci/tmp/build/./src/IO/S3/Client.cpp:828:17 (clickhouse+0x19769e6a) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #4 std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const::'lambda'(DB::S3::UploadPartRequest const&)::operator()(DB::S3::UploadPartRequest const&) const ci/tmp/build/./src/IO/S3/Client.cpp:712:13 (clickhouse+0x19762df5) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #5 std::__1::invoke_result<DB::S3::UploadPartRequest, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequest<DB::S3::UploadPartRequest, std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const::'lambda'(DB::S3::UploadPartRequest const&)>(DB::S3::UploadPartRequest&, DB::S3::UploadPartRequest) const ci/tmp/build/./src/IO/S3/Client.cpp:639:23 (clickhouse+0x19762df5)
E               #6 std::__1::invoke_result<DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0, DB::S3::UploadPartRequest>::type DB::S3::Client::doRequestWithRetryNetworkErrors<false, DB::S3::UploadPartRequest, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0>(DB::S3::UploadPartRequest&, DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const::$_0) const ci/tmp/build/./src/IO/S3/Client.cpp:771:12 (clickhouse+0x19762df5)
E               #7 DB::S3::Client::UploadPart(DB::S3::UploadPartRequest&) const ci/tmp/build/./src/IO/S3/Client.cpp:548:12 (clickhouse+0x19762df5)
E               #8 DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0::operator()() const ci/tmp/build/./src/IO/WriteBufferFromS3.cpp:583:36 (clickhouse+0x198d6d86) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #9 decltype(std::declval<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>(DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x198d6d86)
E               #10 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&>(DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x198d6d86)
E               #11 std::__1::__function::__default_alloc_func<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0, void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x198d6d86)
E               #12 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::WriteBufferFromS3::writePart(DB::WriteBufferFromS3::PartData&&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x198d6d86)
E               #13 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x19506ecf) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #14 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x19506ecf)
E               #15 DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0::operator()() ci/tmp/build/./src/Common/ThreadPoolTaskTracker.cpp:130:9 (clickhouse+0x19506ecf)
E               #16 decltype(std::declval<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>(DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x19506ecf)
E               #17 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&>(DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x19506ecf)
E               #18 std::__1::__function::__default_alloc_func<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0, void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x19506ecf)
E               #19 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::TaskTracker::add(std::__1::function<void ()>&&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x19506ecf)
E               #20 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x155c1162) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #21 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x155c1162)
E               #22 std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()::operator()() ci/tmp/build/./src/Common/threadPoolCallbackRunner.h:45:20 (clickhouse+0x155c1162)
E               #23 decltype(std::declval<void>()()) std::__1::__invoke[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()&>(void&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x155c1162)
E               #24 std::__1::__packaged_task_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'(), std::__1::allocator<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda'()>, void ()>::operator()() ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1438:10 (clickhouse+0x155c1162)
E               #25 std::__1::__packaged_task_function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1589:10 (clickhouse+0x155c1494) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #26 std::__1::packaged_task<void ()>::operator()() ci/tmp/build/./contrib/llvm-project/libcxx/include/future:1752:5 (clickhouse+0x155c1494)
E               #27 std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()::operator()() const ci/tmp/build/./src/Common/threadPoolCallbackRunner.h:52:70 (clickhouse+0x155c13ac) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #28 decltype(std::declval<void>()()) std::__1::__invoke[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&>(void&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x155c13ac)
E               #29 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&>(std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'()&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x155c13ac)
E               #30 std::__1::__function::__default_alloc_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'(), void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x155c13ac)
E               #31 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<std::__1::function<std::__1::future<void> (std::__1::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::__1::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::'lambda'(std::__1::function<void ()>&&, Priority)::operator()(std::__1::function<void ()>&&, Priority)::'lambda0'(), void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x155c13ac)
E               #32 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x1313e25a) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #33 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x1313e25a)
E               #34 ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::worker() ci/tmp/build/./src/Common/ThreadPool.cpp:802:17 (clickhouse+0x1313e25a)
E               #35 decltype(*std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&>().*std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&, void>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x13146cc4) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #36 decltype(auto) std::__1::__apply_tuple_impl[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, 0ul>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<0ul>) ci/tmp/build/./contrib/llvm-project/libcxx/include/tuple:1354:5 (clickhouse+0x13146cc4)
E               #37 decltype(auto) std::__1::apply[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&) ci/tmp/build/./contrib/llvm-project/libcxx/include/tuple:1358:5 (clickhouse+0x13146cc4)
E               #38 ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()::operator()() ci/tmp/build/./src/Common/ThreadPool.h:312:13 (clickhouse+0x13146cc4)
E               #39 decltype(std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)()>()(std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>())) std::__1::__invoke[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x13146be1) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #40 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x13146be1)
E               #41 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>::operator()[abi:ne190107]() ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x13146be1)
E               #42 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x13146be1)
E               #43 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x1313a722) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #44 std::__1::function<void ()>::operator()() const ci/tmp/build/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x1313a722)
E               #45 ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::worker() ci/tmp/build/./src/Common/ThreadPool.cpp:812:17 (clickhouse+0x1313a722)
E               #46 decltype(*std::declval<ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>().*std::declval<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, void>(void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*&&) ci/tmp/build/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x131437db) (BuildId: 399b9b7be8c5c262b642f1cb9faba27b6a809833)
E               #47 void std::__1::__thread_execute[abi:ne190107]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<2ul>) ci/tmp/build/./contrib/llvm-project/libcxx/include/__thread/thread.h:192:3 (clickhouse+0x131437db)
E               #48 void* std::__1::__thread_proxy[abi:ne190107]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>>(void*) ci/tmp/build/./contrib/llvm-project/libcxx/include/__thread/thread.h:201:3 (clickhouse+0x131437db)
```

### How to reproduce

It seems that `LogSeriesLimiter` has non-thread safe `log` method.

--- a/src/IO/S3/Client.cpp
+++ b/src/IO/S3/Client.cpp
@@ -222,7 +222,6 @@ Client::Client(
     , max_redirects(max_redirects_)
     , sse_kms_config(std::move(sse_kms_config_))
     , log(getLogger("S3Client"))
-    , limited_log(std::make_shared<LogSeriesLimiter>(log, 1, 100))
 {
     auto * endpoint_provider = dynamic_cast<Aws::S3::Endpoint::S3DefaultEpProviderBase *>(accessEndpointProvider().get());
     endpoint_provider->GetBuiltInParameters().GetParameter("Region").GetString(explicit_region);
@@ -825,7 +824,7 @@ void Client::slowDownAfterRetryableError() const
         if (current_time_ms >= next_time_ms)
         {
             if (next_time_ms != 0)
-                LOG_TEST(limited_log, "Retry time has passed; proceeding without delay");
+                LOG_TEST(log, "Retry time has passed; proceeding without delay");
             break;
         }
         UInt64 sleep_ms = next_time_ms - current_time_ms;

--- a/src/IO/S3/Client.h
+++ b/src/IO/S3/Client.h
@@ -321,7 +321,6 @@ private:
     const ServerSideEncryptionKMSConfig sse_kms_config;
 
     LoggerPtr log;
-    LogSeriesLimiterPtr limited_log;
 };
 
 class ClientFactory

